src/main/java/net/masterthought/cucumber/json/Match.java

```diff
@@ -1,13 +1,23 @@
 package net.masterthought.cucumber.json;
 
+import com.google.gson.JsonElement;
+
 public class Match {
 
     // Start: attributes from JSON file report
-    private final String location = null;
+    private final JsonElement location = null;
     // End: attributes from JSON file report
 
     public String getLocation() {
-        return location;
+        if (location.isJsonPrimitive() && location.getAsJsonPrimitive().isString()) {
+            return location.getAsString();
+        } else {
+            StringBuilder sb = new StringBuilder();
+            sb.append(location.getAsJsonObject().get("filepath").getAsJsonObject().get("filename").getAsString());
+            sb.append(":");
+            sb.append(location.getAsJsonObject().get("lines").getAsJsonObject().get("data").getAsJsonArray().get(0).getAsString());
```

Comment:
I'm worry if this is safe because:
- name of the attribute "lines" means that it may be more "data" items
- are you sure that "lines" and "data" are both mandatory?
- if you have both line number (from -> to) this information is not compete
- if you add line to the prefix, how would the "Steps" report look like? Lines determine the failure line of the method line?

So, I would left file name only and consider what should be presented by "Steps" report page.